swap/swap_test.go

```diff
@@ -184,267 +184,197 @@ func testBalances(t *testing.T, swap *Swap, expectedBalances map[enode.ID]int64)
-	}
-	sentCheque3, err := swap.SentCheque(testPeer3.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if !reflect.DeepEqual(sentCheque3, generatedCheque3) {
-		t.Fatalf("Expected sent cheque to be %v, but is %v", generatedCheque3, sentCheque3)
-	}
-}
+	sentCheque3 := saveNewSentCheque(t, swap, testPeer3ID)
+	testChequesByPeerAndType(t, swap, map[enode.ID]map[string]*Cheque{testPeerID: {receivedChequeResponseKey: receivedCheque, sentChequeResponseKey: sentCheque}, testPeer2ID: {receivedChequeResponseKey: receivedCheque3, sentChequeResponseKey: sentCheque2}, testPeer3ID: {sentChequeResponseKey: sentCheque3, receivedChequeResponseKey: nil}})
 
-// TestReceivedCheque verifies that received cheques data is correctly obtained
-func TestReceivedCheque(t *testing.T) {
-	// create a test swap account
-	swap, clean := newTestSwap(t, ownerKey, nil)
-	defer clean()
+	// test received cheque for disconnected node
+	receivedCheque4 := saveNewReceivedCheque(t, swap, testPeer3ID)
+	testChequesByPeerAndType(t, swap, map[enode.ID]map[string]*Cheque{testPeerID: {receivedChequeResponseKey: receivedCheque, sentChequeResponseKey: sentCheque}, testPeer2ID: {receivedChequeResponseKey: receivedCheque3, sentChequeResponseKey: sentCheque2}, testPeer3ID: {sentChequeResponseKey: sentCheque3, receivedChequeResponseKey: receivedCheque4}})
 
-	// test cheque addition for peer
-	testPeer, err := swap.addPeer(newDummyPeer().Peer, common.Address{}, common.Address{})
-	if err != nil {
-		t.Fatal(err)
-	}
+	// test cheque change for disconnected node
+	sentCheque4 := saveNewSentCheque(t, swap, testPeer3ID)
+	testChequesByPeerAndType(t, swap, map[enode.ID]map[string]*Cheque{testPeerID: {receivedChequeResponseKey: receivedCheque, sentChequeResponseKey: sentCheque}, testPeer2ID: {receivedChequeResponseKey: receivedCheque3, sentChequeResponseKey: sentCheque2}, testPeer3ID: {sentChequeResponseKey: sentCheque4, receivedChequeResponseKey: receivedCheque4}})
+}
 
-	receivedCheque, err := swap.ReceivedCheque(testPeer.ID())
+// adds a peer to the given Swap struct, fails if there are errors and returns peer otherwise
+func addPeer(t *testing.T, s *Swap) *Peer {
+	t.Helper()
+	peer, err := s.addPeer(newDummyPeer().Peer, common.Address{}, common.Address{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if receivedCheque != nil {
-		t.Fatalf("Expected received cheque to be nil, but is %v", receivedCheque)
-	}
+	return peer
+}
 
-	// generate a random cheque as sent
-	generatedCheque := newRandomTestCheque()
-	err = testPeer.setLastReceivedCheque(generatedCheque)
-	if err != nil {
-		t.Fatal(err)
-	}
-	receivedCheque, err = swap.ReceivedCheque(testPeer.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if receivedCheque != generatedCheque {
-		t.Fatalf("Expected received cheque to be %v, but is %v", generatedCheque, receivedCheque)
-	}
+// generates a cheque and adds it as the last sent cheque for the given peer, fails if there are errors
+func setNewSentCheque(t *testing.T, p *Peer) *Cheque {
+	t.Helper()
```

Comment:
i'm not sure if I see the use of have a test helper that has exactly 1 function call. please remove

Reply:
the use is _only_ readability:

```golang
// test sent cheque for peer
sentCheque := setNewSentCheque(t, testPeer)
```
versus
```golang
// test sent cheque for peer
sentCheque := setNewCheque(t, testPeer.setLastSentCheque) // keep in mind the second parameter here is a function
```
i will admit that i'm not a fan of 1 liner functions. but in this case, i thought it could be justified.

if you insist though, i will remove it.

Reply:
sorry i can't see why the first one is more readable than the second. please remove

Reply:
> also, i don't see why passing `p.setLastSentCheque` is needed. it makes reading the code much more difficult with so many levels of indirection. can't you just call `p.setLastSentCheque(someNewChequeVariable)` from the body of the test?

i can, but it would make the tests much more verbose.

current form:
```golang
sentCheque := setNewSentCheque(t, testPeer)
// snip
sentCheque2 := setNewSentCheque(t, testPeer)
// snip	
sentCheque3 := setNewSentCheque(t, testPeer)
```

removing the function:
```golang
sentCheque := newRandomTestCheque() // this must be saved in a variable in order to be tested later
err := testPeer.setLastSentCheque(sentCheque)
if err != nil {
	t.Fatal(err)
}
// snip
sentCheque2 := newRandomTestCheque() // this must be saved in a variable in order to be tested later
err := testPeer.setLastSentCheque(sentCheque2)
if err != nil {
	t.Fatal(err)
}
// snip	
sentCheque3 := newRandomTestCheque() // this must be saved in a variable in order to be tested later
err := testPeer.setLastSentCheque(sentCheque3)
if err != nil {
	t.Fatal(err)
}
```
passing the function also allows this to be re-used with received cheques, which makes this doubly useful.

i've removed the 1 liners, but kept the `setNewCheque` function. i don't like repeating the same 4 lines of code n times when i can just repeat 1.